firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -115,11 +118,17 @@ public static FirebaseInstallations getInstance(@NonNull FirebaseApp app) {
         .onSuccessTask(this::registerFidIfNecessary);
   }
 
-  /** Returns a auth token(public key) of this Firebase app installation. */
+  /**
+   * Returns a valid authentication token for the Firebase installation. Generates a new token if
+   * one doesn't exist, is expired or about to expire.
+   *
+   * <p>Should only be called if the Firebase Installation is registered.
+   */
   @NonNull
   @Override
-  public Task<InstallationTokenResult> getAuthToken(boolean forceRefresh) {
-    return Tasks.forResult(InstallationTokenResult.builder().build());
+  public Task<String> getAuthToken(final boolean forceRefresh) {
```

Comment:
in general, boolean parameters are a code smell. Could we either split the method (forceGetAuthToken?) or change the param to some sort of enum / int flag here? (See https://developer.android.com/reference/android/support/annotation/IntDef)

Reply:
I used IntDef. PTAL, if I am using it correctly.